File: app/src/main/java/com/example/igreeldownloader/models/FacebookVideo.java
package com.example.igreeldownloader.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class FacebookVideo {
    @SerializedName("error")
    public boolean error;
    @SerializedName("msg")
    public String msg;
    @SerializedName("data")
    public ArrayList<Data> data;

    public static class Data {
        @SerializedName("url")
        public String url;
        @SerializedName("format")
        public String format;
        @SerializedName("ext")
        public String ext;
        @SerializedName("format_id")
        public String format_id;
    }
}

File: app/src/main/java/com/bluelock/realdownloader/models/FacebookVideo.kt
package com.bluelock.realdownloader.models

import com.google.gson.annotations.SerializedName

class FacebookVideo {
    @SerializedName("error")
    var error = false

    @SerializedName("msg")
    var msg: String? = null

    @SerializedName("data")
    var data: ArrayList<Data>? = null

    class Data {
        @SerializedName("url")
        var url: String? = null

        @SerializedName("format")
        var format: String? = null

        @SerializedName("ext")
        var ext: String? = null

        @SerializedName("format_id")
        var format_id: String? = null
    }
}